pkg/applications/providers/template/helm.go

```diff
@@ -96,7 +96,13 @@ func (h HelmTemplate) InstallOrUpgrade(chartLoc string, appDefinition *appskuber
 		return util.NoStatusUpdate, fmt.Errorf("failed to unmarshal values: %w", err)
 	}
 
-	helmRelease, err := helmClient.InstallOrUpgrade(chartLoc, getReleaseName(applicationInstallation), values, *deployOpts, auth)
+	renderedValues, err := h.parsePreDefinedValues(values)
```

Comment:
This function name is super confusing: Why do we have to parse the values we just got from `GetParsedValues()`? Maybe "templateValues" would be a more apt name?